File: pe-math/src/main/java/org/saar/maths/noise/Noise3f.java
package org.saar.maths.noise;

import org.joml.Vector3f;

public interface Noise3f {

    float noise(float x, float y, float z);

    default float noise(Vector3f v) {
        return noise(v.x, v.y, v.z);
    }

}

File: pe-math/src/main/kotlin/org/saar/maths/noise/Noise3f.kt
package org.saar.maths.noise

import org.joml.SimplexNoise
import org.joml.Vector3fc

fun interface Noise3f {
    fun noise(x: Float, y: Float, z: Float): Float

    companion object {
        val simplex = Noise2f(SimplexNoise::noise)
    }
}

fun Noise3f.noise(v: Vector3fc) = noise(v.x(), v.y(), v.z())
